src/dialog_launcher.ts

```diff
@@ -122,95 +135,99 @@ export class Launcher extends search.Search {
                 return ((a_includes && b_includes) || (!a_includes && !b_includes)) ? (a_name > b_name ? 1 : 0) : a_includes ? -1 : b_includes ? 1 : 0;
             });
 
-            // Truncate excess items from the list
-            this.selections.splice(this.list_max());
+            for (const app of apps) this.options.push(app)
```

Comment:
It looks like there's a mix of optional `;` use. Would prettier break the build scripts?